src/protocol/http/src/main/java/org/apache/jmeter/protocol/http/proxy/DefaultSamplerCreator.java

```diff
@@ -359,18 +361,42 @@ protected void computeSamplerName(HTTPSamplerBase sampler,
         String prefix = StringUtils.defaultString(request.getPrefix(), "");
         int httpSampleNameMode = request.getHttpSampleNameMode();
         String format = getFormat(httpSampleNameMode, request.getHttpSampleNameFormat());
+        String url = null;
+        try {
+            url = sampler.getUrl().toString();
+        } catch (MalformedURLException e) {
+            url = "MALFORMED-URL";
+            log.warn("Could not get URL to name sample", e);
+        }
+        List<Object> values = Arrays.asList(
+                prefix,
+                sampler.getPath(),
+                sampler.getMethod(),
+                sampler.getDomain(),
+                sampler.getProtocol(),
+                sampler.getPort(),
+                url
+        );
+        Object[] valuesArray;
         if (!HTTPConstants.CONNECT.equals(request.getMethod()) && isNumberRequests()) {
-            sampler.setName(MessageFormat.format(format, prefix, sampler.getPath(), incrementRequestNumberAndGet()));
+            valuesArray = values.toArray(new Object[values.size() + 1]);
+            valuesArray[values.size()] = incrementRequestNumberAndGet();
         } else {
-            sampler.setName(MessageFormat.format(format, prefix, sampler.getPath()));
+            valuesArray = values.toArray();
```

Comment:
I just wonder: is there a real reason to allow "request number" only in case the method is not CONNECT?
What if we always supply the "request number", and we increase "request number" only in case the method matches?

The dance with `toArray[size+1]` looks strange.

Reply:
To answer your first request (not really), I have no idea, I guess, it was to filter out proxy requests.
About your second concern. I wanted to get around using a full blown ArrayList, but it is probably better to go for readability and use it anyway.